### PR TITLE
doc: AutoML 

### PR DESCRIPTION
**Pull request by ChuyangDeng**
*Issue #, if available:*

*Description of changes:*
Adding documentations to README and automl/README

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -344,3 +345,15 @@ You can use Apache Airflow to author, schedule and monitor SageMaker workflow.
 For more information, see `SageMaker Workflow in Apache Airflow`_.
 
 .. _SageMaker Workflow in Apache Airflow: https://sagemaker.readthedocs.io/en/stable/using_workflow.html
+
+SageMaker Autopilot
+-------------------
+
+Amazon SageMaker Autopilot is an automated machine learning (commonly referred to as AutoML) solution for tabular
```
Comment (laurenyu) [inline README.rst]: I would move "(commonly referred to as AutoML)" to after "solution". Also, put "AutoML" in quotes

File: README.rst [hunk 2]
```diff
@@ -344,3 +345,15 @@ You can use Apache Airflow to author, schedule and monitor SageMaker workflow.
...
+datasets. It automatically trains and tunes the best machine learning models for classification or regression based
+on your data, and host a serials of models on an Inference Pipeline.
```
Comment (laurenyu) [inline README.rst]: * s/host/hosts
* s/serials/series

File: src/sagemaker/automl/README.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
+SageMaker Autopilot
+===================
+
+Amazon SageMaker Autopilot is an automated machine learning (commonly referred to as AutoML) solution for tabular
+datasets. It automatically trains and tunes the best machine learning models for classification or regression based
+on your data, and host a serials of models on an Inference Pipeline.
```
Comment (laurenyu) [inline src/sagemaker/automl/README.rst]: same two comments here

File: src/sagemaker/automl/README.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+SageMaker AutoML Class
+~~~~~~~~~~~~~~~~~~~~~~
+
+The SageMaker ``AutoML`` class is similar to a SageMaker ``Estimator`` where you define the attributes of an AutoML
+job and feed input data to start the job.
+
+A simple use case of an ``AutoML`` object would be:
```
Comment (laurenyu) [inline src/sagemaker/automl/README.rst]: "Here's a simple example of using the `AutoML` object:"

this isn't really a use case as much as it is just usage

File: src/sagemaker/automl/README.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    auto_ml = AutoML(
+        role="sagemaker-execution-role",
+        target_attribute_name="y",
+        sagemaker_session=sagemaker_session,
+    )
+    auto_ml.fit(inputs=inputs)
+
+
+The above code will start an AutoML job (data processing, training, tuning) and output a maximum of 500 candidates by
```
Comment (laurenyu) [inline src/sagemaker/automl/README.rst]: * s/will start/starts
* s/output/outputs

File: src/sagemaker/automl/README.rst [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+default. You can modify the number of output candidates by specifying ``max_candidates`` in the constructor. The AutoML
+job will figure out the problem type (BinaryClassification, MulticlassClassification, Regression), but you can also
+specify the problem type by setting ``problem_type`` in the constructor. Other configurable settings include security
+settings, time limits, job objectives, tags, etc.
```
Comment (laurenyu) [inline src/sagemaker/automl/README.rst]: can we link to ReadTheDocs so people know where to find the rest of the parameters?

File: src/sagemaker/automl/README.rst [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+After an AutoML job is done, there are a few things that you can do with the result.
+
+1. Describe the AutoML job: ``describe_auto_ml_job()`` will give you an overview of the AutoML job, information
```
Comment (laurenyu) [inline src/sagemaker/automl/README.rst]: (optional) in RST, you can use `#.` to have the numbering done automatically

File: src/sagemaker/automl/README.rst [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+job. By calling this method, you can view and compare the candidates.
+
+4. Deploy the best candidate (or any given candidate): ``deploy()`` by default will deploy the best candidate to an
+inference pipeline. But you can also specify a candidate to deploy through ``candidate`` parameter.
+
+
+SageMaker CandidateEstimator Class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The SageMaker ``CandidateEstimator`` class convert a dictionary with AutoML candidate information to an object that
```
Comment (laurenyu) [inline src/sagemaker/automl/README.rst]: s/convert/converts

File: src/sagemaker/automl/README.rst [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+allows you to re-run the candidate's step jobs.
+
+The simplest re-run would be feeding a new dataset but re-use all other configurations that the candidate used to have:
```
Comment (laurenyu) [inline src/sagemaker/automl/README.rst]: * s/would be feeding/is to feed
* s/re-use/reuse
* s/ that the candidate used to have/from the candidate

File: src/sagemaker/automl/README.rst [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+.. code:: python
+
+    candidate_estimator = CandidateEstimator(candidate_dict)
+    inputs = new_inputs
+    candidate_estimator.fit(inputs=inputs)
+
+If you want to have more control over the step jobs of the candidate, you can call ``get_steps()`` and construct
+training/tunning jobs by yourself.
```
Comment (laurenyu) [inline src/sagemaker/automl/README.rst]: s/tunning/tuning

File: README.rst (hunk 1 above)
Comment (ChuyangDeng) [reply README.rst]: Done

File: README.rst (hunk 2 above)
Comment (ChuyangDeng) [reply README.rst]: Updated.

File: src/sagemaker/automl/README.rst (hunk 3 above)
Comment (ChuyangDeng) [reply src/sagemaker/automl/README.rst]: Done.

File: src/sagemaker/automl/README.rst (hunk 5 above)
Comment (ChuyangDeng) [reply src/sagemaker/automl/README.rst]: Done.

File: src/sagemaker/automl/README.rst (hunk 6 above)
Comment (ChuyangDeng) [reply src/sagemaker/automl/README.rst]: I couldn't find the ReadTheDocs link for the AutoML class. For all new classes, only Analytics is there: https://sagemaker.readthedocs.io/en/stable/analytics.html

File: src/sagemaker/automl/README.rst (hunk 8 above)
Comment (ChuyangDeng) [reply src/sagemaker/automl/README.rst]: Done.

File: src/sagemaker/automl/README.rst (hunk 9 above)
Comment (ChuyangDeng) [reply src/sagemaker/automl/README.rst]: Done.

File: src/sagemaker/automl/README.rst (hunk 10 above)
Comment (ChuyangDeng) [reply src/sagemaker/automl/README.rst]: Done.

File: src/sagemaker/automl/README.rst [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+datasets. It automatically trains and tunes the best machine learning models for classification or regression based
+on your data, and hosts a series of models on an Inference Pipeline.
+
+SageMaker AutoML Class
+~~~~~~~~~~~~~~~~~~~~~~
+
+The SageMaker ``AutoML`` class is similar to a SageMaker ``Estimator`` where you define the attributes of an AutoML
+job and feed input data to start the job.
+
+Here's a simple example of using the AutoML ``object``:
```
Comment (laurenyu) [inline src/sagemaker/automl/README.rst]: nit: the backticks should go around "AutoML" not "object"

File: src/sagemaker/automl/README.rst (hunk 6 above)
Comment (laurenyu) [reply src/sagemaker/automl/README.rst]: yeah they have to be added - I've got an open PR to do so: #1161 (won't block this PR on that one, but it'd be nice to follow up on it after)

**Pull request merged**